index.js

```diff
@@ -74,6 +74,20 @@ function parseIntoPreview(data) {
   });
 }
 
+function parseIntoTrackArray(data){
+  if(!data.name)
+    new Error("Data doesn't seem to be of the right shape to parse")
+  if(!data.tracks)
```

Comment:
This seems a bit of a strong assumption, that anything that doesn't have tracks is a track. Maybe its true? What if we checked for data.type === "track" instead?

Or hm, what happends for podcast episodes?